common.sh

```diff
@@ -187,9 +187,29 @@ create_instance()
 get_instance_ip()
 {
     execution_seq=$((${execution_seq}+1))
-    INSTANCE_IPS=$(aws ec2 describe-instances --instance-ids ${INSTANCE_IDS[@]} \
-                        --query "Reservations[*].Instances[*].PrivateIpAddress" \
-                        --output=text)
+    local retry=20
+    local ret=0
+    set +e
+    while [ $retry -ge 0 ]; do
+        INSTANCE_IPS=$(aws ec2 describe-instances --instance-ids ${INSTANCE_IDS[@]} \
+                            --query "Reservations[*].Instances[*].PrivateIpAddress" \
+                            --output=text)
+        ret=$?
+        if [ "$ret" -eq 0 ] && [ -n "$INSTANCE_IPS" ]; then
+            echo "get instance ips successfully."
+            break
+        else
+            echo "describe-instances fails or does not return instance ips, retry after 10 seconds."
+            sleep 10
+        fi
+        retry=$((retry-1))
+    done
+    if [ "$retry" -eq 0 ]; then
+        echo "Failed to get instance ips, exiting ..."
+        exit 1
+    fi
+    set -e
```

Comment:
I really hate our use of set -e through production scripts.  But we should really not blindly re-enable set -e, or we'll have to clean up a ton of code later when we clean up all these scripts.

Reply:
Reading the code, the scripts blindly relies on `set -e` for error checking in most places. So not re-enabling `set -e` could some error ignored in tests after the `get_instance_ip` call. Should I do a refactor for the scripts to do error checking without `set -e` in this PR?

Reply:
https://superuser.com/questions/997755/is-it-possible-to-check-whether-e-is-set-within-a-bash-script

Reply:
I get your point now, I should restore `-e` only if it's previously set, is that correct?

Reply:
that's the least awful approach for now, yes.  It at least does no harm in getting rid of our reliance on -e.

Reply:
Thanks, implemented this in the latest revision.